sys/net/network_layer/fib/fib.c

```diff
@@ -111,11 +111,13 @@ static int fib_find_entry(uint8_t *dst, size_t dst_size,
                 if (fib_table[i].global != NULL) {
                     universal_address_rem(fib_table[i].global);
                     fib_table[i].global = NULL;
+                    fib_table[i].global_flags = 0;
```

Comment:
You're right, they should be cleared unconditionally. I will change that and also add clearing the `iface_id`(I forgot that also).